### ISSUE TITLE
fromCharCode performance issue

### ISSUE THREAD

**Issue by DomiR** [facebook/hermes#141]
So I've noticed that my json export using `react-native-fs` slowed down really bad and I've traced this down to the usage of `https://github.com/mathiasbynens/utf8.js/blob/master/utf8.js` within the `react-native-fs` package (when writing to file with `utf8` encoding).

So the encode function calls the `String.fromCharCode` very often and here you can see the difference between enabling and disabling hermes via the gradle switch.

*With hermes:*
![with_hermes](https://user-images.githubusercontent.com/1834664/67569156-5f3e8580-f72e-11e9-8f36-2d5fd111d93c.gif)

*Without hermes:*
![without_hermes](https://user-images.githubusercontent.com/1834664/67569148-58b00e00-f72e-11e9-9cbd-335dc492a90d.gif)

To replicate this one would probably just need to call fromCharCode in a loop or installing the `utf8` package and encode a large string. 

Logging was done adding:
```
// https://github.com/mathiasbynens/utf8.js/blob/master/utf8.js#L95
if (index % 10000 === 0) {
   console.debug('utf8encode: encoding at %j', index);
}
```

**Comment by dulinriley** [comment]
Were you running a release version of your app? That way it will use the fully-optimized Hermes build.
A small benchmark I made for `String.fromCharCode` shows a large improvement in switching to an opt build (180 ms vs 100 ms).

**Comment by DomiR** [comment]
Yes, unfortunately this happened in the release version as well.

**Comment by dulinriley** [comment]
I have a change going out that should improve the speed of `String.fromCharCode` for a single character to be about twice as fast.

However, in local benchmarks with a tight loop of `String.fromCharCode`, it looks like most of the time is actually spent in the for loop construct itself. Here's an example benchmark:
```
for (var i = 0; i < 1000000; i++) {
  String.fromCharCode(i);
}
```
When running this through `perf`, I see about 60% of the time spent in the interpreter loop, and 5% of the time in `stringFromCharCode` (after my optimization is applied).

**Comment by dulinriley** [comment]
Also, for the loop in utf8.js you mention, it seems to be building up a string one character at a time.
This is slower in Hermes right now than most other JS engines, but there's a fix in progress.
You can track progress on that performance improvement in issue #111
